apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx

```diff
@@ -42,211 +49,239 @@ yarn add @supabase/auth-helpers-sveltekit
 
-To make the session available to the UI (pages, layouts), pass the session in the root layout server load function:
+In order to make the session available to the UI (pages, layouts) we need to pass the session in the root layout server load function:
 
 ```ts title=src/routes/+layout.server.ts
-import type { LayoutServerLoad } from './$types'
-import { getServerSession } from '@supabase/auth-helpers-sveltekit'
+// src/routes/+layout.server.ts
+import type { LayoutServerLoad } from './$types';
 
-export const load: LayoutServerLoad = async (event) => {
+export const load: LayoutServerLoad = async ({ locals: { getSession } }) => {
   return {
-    session: await getServerSession(event),
-  }
-}
+    session: getSession()
+  };
+};
 ```
 
-In addition you can create a layout load function if you are using `invalidate('supabase:auth')`:
+#### Shared Load functions and pages
 
-```ts title=src/routes/+layout.ts
-import type { LayoutLoad } from './$types'
-import { getSupabase } from '@supabase/auth-helpers-sveltekit'
+To be able to use supabase in shared load functions and inside pages you need to create a supabase client in the root layout load:
 
-export const load: LayoutLoad = async (event) => {
-  const { session } = await getSupabase(event)
-  return { session }
-}
+```ts
+// src/routes/+layout.ts
+import { invalidate } from '$app/navigation';
+import {
+  PUBLIC_SUPABASE_ANON_KEY,
+  PUBLIC_SUPABASE_URL
+} from '$env/static/public';
+import { createSupabaseLoadClient } from '@supabase/auth-helpers-sveltekit';
+import type { LayoutLoad } from './$types';
+import type { Database } from '../DatabaseDefinitions';
+
+export const load: LayoutLoad = async ({ fetch, data, depends }) => {
+  depends('supabase:auth');
+
+  const supabase = createSupabaseLoadClient<Database>({
+    supabaseUrl: PUBLIC_SUPABASE_URL,
+    supabaseKey: PUBLIC_SUPABASE_ANON_KEY,
+    event: { fetch },
+    serverSession: data.session,
+    onAuthStateChange() {
+      invalidate('supabase:auth');
+    }
+  });
+
+  const {
+    data: { session }
+  } = await supabase.auth.getSession();
+
+  return { supabase, session };
+};
 ```
 
-This results in fewer server calls as the client manages the session on its own.
+The client can be accessed inside pages by `$page.data.supabase` or `data.supabase` when using `export let data: PageData`.
 
-### Typings
+The usage of `depends` and `invalidate` tells sveltekit that this load function should be executed whenever the session updates to keep the page store in sync.
 
-In order to get the most out of TypeScript and it´s intellisense, you should import our types into the `app.d.ts` type definition file that comes with your SvelteKit project.
+`createSupabaseLoadClient` caches the client when running in a browser environment and therefore does not create a new client for every time the load function runs.
 
-```ts title=src/app.d.ts
-/// <reference types="@sveltejs/kit" />
+### Typings with v2 from Supabase CLI
```

Comment:
"Generate types from your database"  ?

Note that you can also download them right from the table now (API Preview) button